android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt

```diff
@@ -36,35 +45,81 @@ class AnotherLocationPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
                 permissionEventsSource = null
             }
         }
-    private lateinit var context: Context
-    private lateinit var activity: Activity
 
+    private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    private lateinit var locationRequest: LocationRequest
+    private lateinit var locationCallback: LocationCallback
+
+    private lateinit var locationEventsEmitter: EventChannel
+    private var locationEventsSource: EventChannel.EventSink? = null
+    private var locationEventsSourceHandler: EventChannel.StreamHandler =
+        object : EventChannel.StreamHandler {
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+                locationEventsSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+                locationEventsSource = null
+            }
+        }
 
     override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
         channel = MethodChannel(flutterPluginBinding.binaryMessenger, PLUGIN_CHANNEL_NAME)
         channel.setMethodCallHandler(this)
         context = flutterPluginBinding.applicationContext
+
         permissionEventsEmitter =
             EventChannel(flutterPluginBinding.binaryMessenger, PERMISSION_CHANNEL_NAME)
         permissionEventsEmitter?.setStreamHandler(permissionEventsSourceHandler)
+
+        locationEventsEmitter =
+            EventChannel(flutterPluginBinding.binaryMessenger, LOCATION_CHANNEL_NAME)
+        locationEventsEmitter?.setStreamHandler(locationEventsSourceHandler)
     }
 
     override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
         when (call.method) {
             GET_PLATFORM_VERSION -> getPlatformVersion(result)
             CHECK_PERMISSIONS -> checkPermissions()
             REQUEST_PERMISSIONS -> requestPermissions()
+            INITIALIZE_PLUGIN -> initializePlugin()
+            GET_LOCATION -> getLocation()
+            STOP_PLUGIN -> abortGetLocation()
             else -> result.notImplemented()
         }
     }
 
+    private fun initializePlugin() {
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(activity)
+        val hasPermissions = checkPermissions()
+        if (hasPermissions) {
+            locationRequest = create().apply {
+                interval = INTERVAL
+                priority = PRIORITY_HIGH_ACCURACY
+                locationCallback = object : LocationCallback() {
+                    override fun onLocationResult(result: LocationResult?) {
+                        locationEventsSource?.success(
+                            LONGITUDE_TAG
+                                    + " " + result?.lastLocation?.longitude.toString()
+                                    + "  " + LATITUDE_TAG
+                                    + " " + result?.lastLocation?.latitude.toString()
+                        )
```

Comment:
Insted to do this you can use a StringBuilder, to generate the correct concatenation

Reply:
Done!